docs/updates/5-1-2017-flow.md

```diff
@@ -0,0 +1,156 @@
+### Flow Coverage Report (May 1st)
+
+`yarn flow-coverage`
+
+Stats:
+* Flow version `0.43.1`
+* **60 %**
+
+#### Components
+
+* **54%** coverage
+* 44 components are typed (almost all)
+* `yarn flow-react` - creates a flow report for components
+
+Typing our components helps us with rendering logic, which can take advantage
+of our domain objects such as `Frame`.
+
+```js
+function renderFrameTitle(frame: Frame) {
+  const displayName = formatDisplayName(frame);
+  return dom.div({ className: "title" }, displayName);
+}
+```
+
+We are not yet typing our props and state, which would be a helpful way
+to get additional coverage. We currently use `PropTypes`, which do not include
+flow types. We will hopefully switch to [props] soon to take advantage of static typing.
+
+```js
+shouldComponentUpdate(nextProps, nextState) {
+  const { frames, selectedFrame } = this.props;
+  const { showAllFrames } = this.state;
+  return (
+    frames !== nextProps.frames ||
+    selectedFrame !== nextProps.selectedFrame ||
+    showAllFrames !== nextState.showAllFrames
+  );
+}
+```
+
+There have been a couple places where flow types have been difficult to get right.
+
+1. **contravariants** `this.toggleFramesDisplay = this.toggleFramesDisplay.bind(this)`
```

Comment:
@thejameskyle Great explanation, was wondering about why this happens myself too.

Is it the same case when I try to set custom `toString()` on some function?
```javascript
const createActionCreator = (type, actionCreator) = {
  actionCreator.toString = () => type // error!
  return actionCreator
}
```
is there any workaround for this?